components/brave_wallet_ui/components/desktop/views/explore_web3/explore_web3.tsx

```diff
@@ -64,6 +56,11 @@ export const ExploreWeb3View = () => {
   // state
   const [searchValue, setSearchValue] = React.useState<string>('')
   const [showFilters, setShowFilters] = React.useState<boolean>(false)
+  const [selectedDapp, setSelectedDapp] = React.useState<
+    BraveWallet.Dapp | undefined
+  >(undefined)
+  const [isDappDetailsOpen, setIsDappDetailsOpen] =
+    React.useState<boolean>(false)
```

Comment:
Is this state needed? Wont the modal always be open if the selected dapp is not `undefined`